sdks/python/apache_beam/ml/inference/base.py

```diff
@@ -96,7 +107,9 @@ def expand(self, pcoll: beam.PCollection) -> beam.PCollection:
         pcoll
         # TODO(BEAM-14044): Hook into the batching DoFn APIs.
         | beam.BatchElements()
-        | beam.ParDo(_RunInferenceDoFn(self._model_loader, self._clock)))
+        | beam.ParDo(
+            _RunInferenceDoFn(
+                self._model_loader, self._prediction_params, self._clock)))
```

Comment:
This is also what I preferred. But after a long discussion we settled on making these a side input.

The reason was that since these static parameters were conceptually different because the were parameters for the inputs rather than model loading parameters, they should not be passed in on the model loader. If you want to rehash this I suggest setting up a meeting.

Reply:
Note: The code has been changed slightly to `**kwargs` instead of `self._prediction_params`.

Passing in these extra prediction-related parameters as side inputs makes more conceptual sense because both the `batch` examples and `prediction_params` kwargs are different types of inputs that are both passed at the same time into Pytorch model's predict call (which are both passed into its `forward` function)
```
# Inside PytorchInferenceRunner()
def run_inference(batch, model, **kwargs):
   prediction_params = kwargs.get('prediction_params')
   model(batch, **prediction_params)

# this call goes inside Pytorch model ...
class PytorchModel(nn.Module):
    def forward(self, batch, **prediction_params):
        ...
```

Reply:
> But after a long discussion we settled on making these a side input.

This is not a side input, this is a constant parameter. A side input comes from a PCollection wrapped in a PCollectionView and is passed to the ParDo: https://beam.apache.org/documentation/programming-guide/#side-inputs

That's what I'm trying to suggest, making it a side input would justify a change to base RunInference, because it means you'd need to change the `ParDo` creation.

Reply:
Thanks for the clarification Brian. 

It may be the case that these extra `prediction_params` need to be determined at runtime (not hard-coded) based on, for example, the shape of the input data. So I think we'd want to modify base.. I'm going to do a few quick tests right now to see the flow of passing in `prediction_params` as a PCollection.